buildkite/resource_cluster_agent_token.go

```diff
@@ -80,6 +87,10 @@ func (ct *clusterAgentToken) Schema(_ context.Context, _ resource.SchemaRequest,
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"allowed_ip_addresses": resource_schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "A list of CIDR-notation IPv4 addresses from which agents can use this Cluster Agent Token.",
+			},
```

Comment:
I think we should allow these to be an actual separate list instead of space/comma separated. Would integrate a bit nicer with other providers that return lists (and one of our existing IP attributes works the same way as well)

Reply:
So in state its a list of strings and we join it manually for the API requests